Fix PHPCS rules to correctly space arrays and function calls [MAILPOET-5946]

mikejolley opened the pull request:
## Description

Fixes PHPCS config to enforce the following spacing rules:

```php
[0 => 'value', 1 => 1]; 
call($argument1, $argument2);
```

And fixes cases like this:

```php
[ 0 => 'value', 1 => 1 ]; // space after opening and closing
[0 => 'value',1 => 1]; // no space after comma
call( $argument1, $argument2 ); // space after opening and closing
call($argument1,$argument2) // no space after comma
```

After fixing the rules I ran phpcbf to fix violations:

```
./tasks/code_sniffer/vendor/bin/phpcbf ./lib --standard=./tasks/code_sniffer/MailPoet/free-ruleset.xml
```

## Code review notes

I tested this in VS Code using `obliviousharmony.vscode-php-codesniffer` extension and the following workspace config:

```json
{
  "phpCodeSniffer.standard": "Custom",
  "phpCodeSniffer.standardCustom": "mailpoet/tasks/code_sniffer/MailPoet/free-ruleset.xml",
  "phpCodeSniffer.exec.osx": "mailpoet/tasks/code_sniffer/vendor/bin/phpcs"
}
```

Then added this code to any PHP file and saved:

```php
$test = [ 0 => 'value', 1 => 1 ]; // space after opening and closing
$test = [0 => 'value',1 => 1]; // no space after comma
$test = call( $argument1, $argument2 ); // space after opening and closing
$test = call($argument1,$argument2); // no space after comma
```

After saving check the spacing is corrected.

## QA notes

This PR applies phpcbf rules and shouldn't impact any existing functionality.

## Linked PRs

_N/A_

## Linked tickets

[MAILPOET-5946](https://mailpoet.atlassian.net/browse/MAILPOET-5946)

## After-merge notes

We need to apply phpcbf to the premium plugin also.

## Tasks

- [ ] I followed [best practices](https://codex.wordpress.org/I18n_for_WordPress_Developers) for translations
- [ ] I added sufficient test coverage
- [ ] I embraced TypeScript by either creating new files in TypeScript or converting existing JavaScript files when making changes


[MAILPOET-5946]: https://mailpoet.atlassian.net/browse/MAILPOET-5946?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

mikejolley committed:
Update sniff packages

mikejolley committed:
Update rules to include correct spacing around arrays and closures.

mikejolley committed:
Apply rules on lib dir

mikejolley committed:
Apply rules to tests and tasks

mikejolley committed:
Fix minimal standard in composer

mikejolley commented:
@JanJakes Rostislav suggested I request a review from you. Let me know if I've missed anything. Since this touches many files it could get stale fairly quickly.

JanJakes commented on mailpoet/tasks/code_sniffer/composer.json:
```diff
@@ -8,7 +8,7 @@
     "squizlabs/php_codesniffer": "^3.9",
     "wporg/plugin-directory": "dev-trunk",
-    "wp-coding-standards/wpcs": "^3.0"
+    "wp-coding-standards/wpcs": "^2.0"
```
Did you have to downgrade here for some reason?

mikejolley replied:
Yes there was an incompatibility with the `wporg/plugin-directory` package where we take the "minimum" plugin rules. I think we may want to replace that and upgrade at some point but for the sake of getting this passing I downgraded it.

JanJakes commented on mailpoet/tasks/code_sniffer/composer.json:
```diff
@@ -18,7 +18,7 @@
   "config": {
     "platform": {
-      "php": "7.1.8"
+      "php": "8.1"
```
Do you think we could go with a minimum version requirement here (like `>= 8.0` or something) or does it need to be exactly `8.1`?

mikejolley replied:
`slevomat/coding-standard` requires > 8.0 so I had to bump this.

`>=` doesn't appear to work here, and the previous version was specifically defined. So I think its ok to define 8.1. I know 8.1 works anyway!

JanJakes reviewed (commented):
@mikejolley Thanks for this! We had so many inconsistencies 🙀 Tried it locally, works great!

One thing to note is that this config is reused by the [premium plugin](https://github.com/mailpoet/mailpoet-premium), so we'll need to run `phpcbf` there as well. I'm not sure if you have premium set up; so in case you want me to run the fixes there, just let me know.

mikejolley committed:
Spacing

mikejolley commented:
@JanJakes I've rebased and fixed new issues and responded to comments. Just need your approval. If it needs QA approval too I'll probably have to rebase again when they're done. I imagine this will get stale quick.

> One thing to note is that this config is reused by the [premium plugin](https://github.com/mailpoet/mailpoet-premium), so we'll need to run phpcbf there as well. I'm not sure if you have premium set up; so in case you want me to run the fixes there, just let me know.

Would we run this after merging this PR or do they both need to happen at the same time? I've not contributed to the premium version yet, but if someone is on hand to guide me through it I can run it on there when this PR is done.

JanJakes commented:
@mikejolley

> Would we run this after merging this PR or do they both need to happen at the same time? I've not contributed to the premium version yet, but if someone is on hand to guide me through it I can run it on there when this PR is done.

I also think we should merge this PR soon, so that it doesn't become stale, but with the premium plugin the issue is that the CI builds (after the free PR is merged) will start to fail.

So our usual process is to create both PRs at the same time — when the branch name is the same, the premium plugin on CI will use the corresponding free plugin branch (this one), and it will all pass. Then we link them in the PR description, and review + merge them both together.

Setting up the premium plugin [should happen as part of the initial setup](https://github.com/mailpoet/mailpoet?tab=readme-ov-file#-initial-setup), but if not, it shouldn't be a big deal — it just needs to be checked out to the root of the mailpoet monorepo (see [this line](https://github.com/mailpoet/mailpoet/blob/trunk/dev/initial-setup.sh#L4)), it needs an `.env` file, similar to the one for the free plugin (see [here](https://github.com/mailpoet/mailpoet/blob/5ec6cc1280e6bce1717c9bebc127bf85dcf58ef9/dev/initial-setup.sh#L16)), and then for the Docker, that may needs to be restarted for the volume to mount.

Then you should see the premium plugin in your development site, and you can just activate it, when needed. For the CI commands, for CodeSniffer it should be the same `./do` command, just under `mailpoet/mailpoet-premium`.

If there's any trouble, or you're busy with other stuff, don't hesitate to ping me or anyone else to help with the setup or run these commands in the premium plugin.

mikejolley commented:
@JanJakes I think I've done this correct. See https://github.com/mailpoet/mailpoet-premium/pull/854

JanJakes commented:
@mikejolley Awesome, thank you!

Merging, as this needs no QA.
